### PR TITLE
Reader: Enable Follow Conversation for DotCom and Jetpack sites

### LINKED ISSUE #15273: P2: Support Follow / Unfollow conversations button on DotCom and Jetpack sites
Background: https://github.com/wordpress-mobile/WordPress-iOS/issues/14883

~As an early adopter of P2s, I want to follow and unfollow conversations for internal P2s, wherever in the Reader that I happen to find them.~

Ref. pbArwn-1n5-p2 

Enable the "Follow Conversation" feature for all DotCom and Jetpack-connected sites. This feature does not exist for external (self-hosted, no Jetpack) sites.

Currently the post topic is set manually and changes when the user navigates to a different Reader tab (going to the Following tab, for example). We will need more robust settings to that older internal P2s can be identified outside of the Automattic tab.

- [x] Reader > Following tab > Post > Comment buttons should show the follow / unfollow button.
- [x] Reader > Automattic > Post Header tapped > Post > Comment buttons should show the follow / unfollow button.

Currently implemented:
- [x] Reader > Automattic > Post > Comment buttons display the follow / unfollow button.
- [x] Reader > Following > (Any new P2 containing the `is_wpforteams_site`) Post > Comment buttons > display the follow / unfollow button.

##### Tested on iPhone 11 Pro, iOS 13.7, WPiOS 16.1

### PR DESCRIPTION
**Pull request by mindgraffiti**
Ref. pbArwn-1n5-p2
Closes #15273 

This PR enables the "Follow Conversation" feature for all DotCom and Jetpack sites. This feature does not exist for external sites.

### To test
**Set up**
On the web or in the mobile app, ensure you have followed:
- a DotCom or Jetpack site
- an external site (self-hosted, not connected to Jetpack)

So that the "Following" tab inside of Reader is populated with posts.

**Test Case 1**
Follow Conversation on a DotCom or Jetpack site.

1. Navigate to Reader > Following tab.
2. Find a post where the Comments button is enabled.
3. Tap the Comments button.
4. In the Reader Comments view, ensure the Follow Conversation button is displayed.
5. Optional: subscribe / unsubscribe to see the button states and toast messages.

**Test Case 2**
1. Navigate to Reader > Following tab.
2. Find a post from an external site.
3. Notice the Comment button is disabled.

### Screenshots
| iPhone 11 Pro | iOS 14.2 |
| --- | --- |
| **Reader > Following** | **Comments view** |
|  Simulator Screen Shot - iPhone 11 Pro - 2020-12-02 at 09 40 09 | Simulator Screen Shot - iPhone 11 Pro - 2020-12-02 at 08 51 44 |
| **Following** | **Unfollowing** | 
| Simulator Screen Shot - iPhone 11 Pro - 2020-12-02 at 09 58 30 | Simulator Screen Shot - iPhone 11 Pro - 2020-12-02 at 09 58 40 |

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by mindgraffiti** [comment]
Error messages conform to Android https://github.com/wordpress-mobile/WordPress-Android/pull/13495

File: WordPress/Classes/Services/FollowCommentsService.swift [hunk 1]
```diff
@@ -27,17 +27,12 @@ class FollowCommentsService: NSObject {
...
-    /// Note that blogs in the Reader's Automattic tab are only available to Automatticians.
     @objc var canFollowConversation: Bool {
-        guard let blog = post.blog else {
-            return false
+        if post.isJetpack || post.isWPCom {
+            return true
         }
-        return blog.isWPForTeams() || ReaderHelpers.topicType(post.topic) == .team
+
+        return false
```
Comment (ScoutHarris) [inline WordPress/Classes/Services/FollowCommentsService.swift]: Couldn't this just be `return post.isJetpack || post.isWPCom`?

File: WordPress/Classes/ViewRelated/Reader/ReaderCommentsViewController.m [hunk 2]
```diff
@@ -1235,8 +1235,8 @@ - (void)handleFollowConversationButtonTapped
         DDLogError(@"Error toggling subscription status: %@", error);
 
         NSString *title = newIsSubscribed
-            ? NSLocalizedString(@"There has been an unexpected error while subscribing to the comments", "The app failed to subscribe to the comments for the post")
-            : NSLocalizedString(@"There has been an unexpected error while unsubscribing from the comments", "The app failed to unsubscribe from the comments for the post");
+            ? NSLocalizedString(@"Could not subscribe to comments", "The app failed to subscribe to the comments for the post")
+            : NSLocalizedString(@"Could not unsubscribe from comments", "The app failed to unsubscribe from the comments for the post");
```
Comment (ScoutHarris) [inline WordPress/Classes/ViewRelated/Reader/ReaderCommentsViewController.m:1239]: 👍

**Review by ScoutHarris — APPROVED** [review]
Nice! Works as expected. Just one nitpick. Otherwise :shipit: !

File: WordPress/Classes/Services/FollowCommentsService.swift (hunk 1 above)
Comment (mindgraffiti) [reply WordPress/Classes/Services/FollowCommentsService.swift]: oh yea! I like how simplified that is

**Comment by mindgraffiti** [comment]
Thank you for the review @ScoutHarris 🙇‍♀️

**Pull request merged**